src/Components/Components/src/Auth/AuthenticationStateProvider.cs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Components
+{
+    /// <summary>
+    /// Provides information about the authentication state of the current user.
+    /// </summary>
+    public abstract class AuthenticationStateProvider
+    {
+        /// <summary>
+        /// Gets an <see cref="AuthenticationState"/> instance that describes
+        /// the current user.
+        /// </summary>
+        /// <returns>An <see cref="AuthenticationState"/> instance that describes the current user.</returns>
+        public abstract Task<AuthenticationState> GetAuthenticationStateAsync();
+
+        /// <summary>
+        /// An event that provides notification when the <see cref="AuthenticationState"/>
+        /// has changed. For example, this event may be raised if a user logs in or out.
+        /// </summary>
+#pragma warning disable 0067 // "Never used" (it's only raised by subclasses)
+        public event AuthenticationStateChangedHandler AuthenticationStateChanged;
+#pragma warning restore 0067
```

Comment:
Is this actually needed now that `NotifyAuthenticationStateChanged` is there?

Reply:
Well spotted. Rather than go through another multi-hour CI process to make this tweak, I'm going to include it in my next auth PR.